[DOCS-14989] Break up EUDM setup guide into pages per OS (#38311)
DataDog/documentation: content/en/infrastructure/end_user_device_monitoring/setup/macos.md

---
title: Set up End User Device Monitoring on macOS
description: Deploy the Datadog Agent on macOS devices for End User Device Monitoring.
further_reading:
   - link: "/infrastructure/end_user_device_monitoring/"
     tag: "Documentation"
     text: "End User Device Monitoring"
   - link: "/infrastructure/end_user_device_monitoring/setup/"
     tag: "Documentation"
     text: "Set up End User Device Monitoring"
---

{{< callout url="https://www.datadoghq.com/product-preview/end-user-device-monitoring/" btn_hidden="false" >}}
End User Device Monitoring is in Preview. To enroll, click <b>Request Access</b>.
{{< /callout >}}

There are two ways to deploy the Datadog Agent on macOS devices for End User Device Monitoring. Choose the method that fits your environment:

- **Manual install**: Run a one-line script directly on the device. Best for testing a single machine or devices not managed by a mobile device management (MDM) solution.
- **MDM deploy**: Push the Agent to a fleet of managed Macs. This example uses a Jamf Pro policy and shell script to install the Agent in the background with no interaction on each device.

> Regardless of the method, the installer creates a dedicated system user named `_dd-agent`. This is expected: it allows the Agent to run as a background service with the appropriate system permissions. You do not need to manage this user.

Reviewer: ```suggestion
Regardless of the method, the installer creates a dedicated system user named `_dd-agent`. This allows the Agent to run as a background service with the appropriate system permissions. You do not need to manage this user.
```

<div class="alert alert-danger">The <code>infrastructure_mode: end_user_device</code> setting is required. Without it, the device does not appear in the End User Devices view and is billed as a host.</div>

{{< tabs >}}
{{% tab "Manual install" %}}

> Use this method to test a single machine or for devices that are not managed by an MDM solution.

Reviewer: ```suggestion
Use this method to test a single machine, or for devices that are not managed by an MDM solution.
```

### Prerequisites

- Administrator (`sudo`) access on the Mac.
- A valid Datadog API key. For instructions, see [API and Application Keys][101].
- macOS 11 or later.
- An active internet connection on the device.

### Install the Agent

1. In Datadog, navigate to [{{< ui >}}Fleet Automation{{< /ui >}} > {{< ui >}}Install Agents{{< /ui >}} > {{< ui >}}macOS{{< /ui >}}][102].
1. Toggle on {{< ui >}}Enable End User Device Monitoring{{< /ui >}}.
1. Click {{< ui >}}Select API Key{{< /ui >}} and choose an API key.
1. Copy the provided installation command beginning with `sudo DD_API_KEY=`.

> 1. Open {{< ui >}}Terminal{{< /ui >}}. Find it in {{< ui >}}Applications{{< /ui >}} > {{< ui >}}Utilities{{< /ui >}} > {{< ui >}}Terminal{{< /ui >}}, or search for it with Spotlight (<kbd>⌘</kbd> + <kbd>Space</kbd>).

Reviewer: This is documenting 3rd party (MacOS) functionality - i'm not considering it a blocker, but this step would be fine without it (i.e. you can just say "Open the system Terminal" or something similar), and it adds maintenance overhead if the procedure ever changes or diverges in a future OS update, etc.

Author: Yeah, that's a good point! This was in the original but I agree it's not great practice to leave it in. I'll remove

> 1. Paste the installation command into the terminal and run it. Enter your Mac password when prompted.

Author: ```suggestion
1. Open Terminal, paste the installation command in and run it. Enter your Mac password when prompted.
```

    The script installs the Agent, creates the `_dd-agent` system user, and registers the Agent as a background launch daemon. Installation takes one to two minutes.

### Verify the installation

> To confirm that the Agent is running, in Datadog, go to [**Infrastructure** > **End User Devices**][104]. Your device appears within 5-10 minutes. If it does not appear after 10 minutes, verify your API key and confirm that the configuration was saved and the Agent was restarted.

Reviewer: ```suggestion
To confirm that the Agent is running, in Datadog, go to [**Infrastructure** > **End User Devices**][104]. Your device appears within 5-10 minutes. If it does not appear after 10 minutes, verify your API key, confirm the configuration, and restart the Agent.
```

Rewording a bit to remove passive voice.

> Alternatively, you can run the following command in Terminal to verify the installation:

Reviewer: ```suggestion
Alternatively, run the following command in Terminal to verify the installation:
```

```shell
sudo datadog-agent status
```

In the output, confirm the following:
- The Agent version is 7.80 or later.
- **Status** is `Running`.
- `infrastructure_mode: end_user_device` is set.

[101]: /account_management/api-app-keys/
[102]: https://app.datadoghq.com/fleet/install-agent/latest?platform=macos
[103]: /agent/supported_platforms/osx/#commands
[104]: https://app.datadoghq.com/end-user-devices

{{% /tab %}}

> {{% tab "MDM deploy (example with Jamf Pro)" %}}

Reviewer: The parenthetical seems unnecessary/out of place in the tab title, i recommend removing it.

Author: This was a request from @eho1307 to make it extra clear that you can use other MDMs, not just Jamf Pro. Erica, let me know what you think

This example method uses a Jamf Pro policy and shell script to install and configure the Datadog Agent across your fleet of managed Macs, with no manual steps on each device. The script performs the same actions as the manual install, but runs in the background when the policy triggers.

> You can use other MDMs; however, this page walks through Jamf Pro as an example.

Reviewer: ```suggestion
You can use other MDMs; however, this procedure walks through Jamf Pro as an example.
```

1. Click {{< ui >}}Select API Key{{< /ui >}} and choose an API key.
1. Copy the provided installation command beginning with `sudo DD_API_KEY=`.
1. Open {{< ui >}}Terminal{{< /ui >}}. Find it in {{< ui >}}Applications{{< /ui >}} > {{< ui >}}Utilities{{< /ui >}} > {{< ui >}}Terminal{{< /ui >}}, or search for it with Spotlight (<kbd>⌘</kbd> + <kbd>Space</kbd>).
1. Paste the installation command into the terminal and run it. Enter your Mac password when prompted.

    The script installs the Agent, creates the `ddagent` system user, and registers the Agent as a background launch daemon. Installation takes one to two minutes.

1. Launch the Datadog Agent Manager UI by running:

    ```shell
    sudo datadog-agent launch-gui
    ```

1. Navigate to {{< ui >}}Settings{{< /ui >}}.
1. Add a new parameter to the end of the YAML file:

    ```yaml
    infrastructure_mode: end_user_device
    ```

1. Save your changes, then [restart the Agent][2].

### Verify the installation

1. To confirm that the Agent is running, run the following command in Terminal:

    ```shell
    sudo datadog-agent status
    ```

    In the output, confirm the following:

>     - The Agent version is 7.80 or later.

Author: @eho1307 I went with 7.80 but we could also change this to a more generic latest if you prefer:

```suggestion
    - The Agent version is the latest available version.
```

- macOS 11 or later.
- An active internet connection on the device.

### Install the Agent

1. In Datadog, navigate to [{{< ui >}}Fleet Automation{{< /ui >}} > {{< ui >}}Install Agents{{< /ui >}} > {{< ui >}}macOS{{< /ui >}}][6].
1. Click {{< ui >}}Select API Key{{< /ui >}} and choose an API key.
1. Copy the provided installation command beginning with `sudo DD_API_KEY=`.
1. Open {{< ui >}}Terminal{{< /ui >}}. Find it in {{< ui >}}Applications{{< /ui >}} > {{< ui >}}Utilities{{< /ui >}} > {{< ui >}}Terminal{{< /ui >}}, or search for it with Spotlight (<kbd>⌘</kbd> + <kbd>Space</kbd>).
1. Paste the installation command into the terminal and run it. Enter your Mac password when prompted.

    The script installs the Agent, creates the `ddagent` system user, and registers the Agent as a background launch daemon. Installation takes one to two minutes.

1. Launch the Datadog Agent Manager UI by running:

    ```shell
    sudo datadog-agent launch-gui
    ```

1. Navigate to {{< ui >}}Settings{{< /ui >}}.
1. Add a new parameter to the end of the YAML file:

    ```yaml
    infrastructure_mode: end_user_device
    ```

> 1. Save your changes, then [restart the Agent][2].

Reviewer 2: Can we add the specific command here?

Author: Done! I left the link too but updated it to point to a different page (the only I was linking to originally only covers Windows/Linux)

### Verify the installation

1. To confirm that the Agent is running, run the following command in Terminal:

    ```shell
    sudo datadog-agent status
    ```

    In the output, confirm the following:
    - The Agent version is 7.80 or later.
    - **Status** is `Running`.
    - `infrastructure_mode: end_user_device` is set.

1. In Datadog, go to [**Infrastructure** > **End User Devices**][4]. Your device appears within 5-10 minutes. If it does not appear after 15 minutes, verify your API key and confirm that the configuration was saved and the Agent was restarted.

### Enable Network Path (optional)

> 1. Add the following to `/opt/datadog-agent/etc/system-probe.yaml`:

Reviewer 2: Not sure why this isn't appearing in the preview link

Author: Ah this is content I just added, so it should show up in the next preview build (20 minutes or so)

    ```yaml
    traceroute:
      enabled: true
    ```

> 1. Make a copy of `/opt/datadog-agent/etc/conf.d/network_path.d/conf.yaml.example`, removing `.example` from the filename:

Author: @jack0x2 the path in the google doc starts with `/etc` but I think the Agent installs under `/opt` on Mac. Let me know if this is wrong!

Reviewer 3: yes it is `/opt/datadog-agent/etc/conf.d/network_path.d/`

        ## Traceroute method used to monitor an endpoint via Network Path
        ## Available methods: syn, sack, prefer_sack, syn_socket (syn_socket only works on Windows OSes)
        ## Note: Windows client versions only support syn_socket
        #
        # tcp_method: <METHOD>

        ## @param traceroute_queries - integer - optional - default: 3
        ## Number of traceroutes to send for each check run.
        #
        # traceroute_queries: 3

        ## @param e2e_queries - integer - optional - default: 50
        ## Number of end-to-end probes to send for each check run.
        #
        # e2e_queries: 50

        ## @param tags - list of strings - optional
        ## A list of tags to attach to every metric and service check emitted by this instance.
        ##
        ## Learn more about tagging at https://docs.datadoghq.com/tagging
        #
        # tags:
        #   - <KEY_1>:<VALUE_1>
        #   - <KEY_2>:<VALUE_2>
    ```

1. Restart the Agent and system probe to start seeing network paths:

    ```shell
    sudo launchctl kickstart -k system/com.datadoghq.agent
    sudo launchctl kickstart -k system/com.datadoghq.sysprobe
    ```

> ## MDM deploy with Jamf Pro (multiple devices)

Reviewer 2: Can we make this a tabbed element?

Author: Do you mean have the manual install vs. MDM deploy be in tabs? Or do you want to add other examples besides Jamf in tabs underneath the MDM section?

Reviewer 2: yeah manual vs mdm in tabs

Author: Added! Let me know what you think in the preview

> This method uses a Jamf Pro policy and shell script to install and configure the Datadog Agent across your fleet of managed Macs, with no manual steps on each device. The script performs the same actions as the manual install, but runs in the background when the policy triggers.

Reviewer 2: ```suggestion
This example method uses a Jamf Pro policy and shell script to install and configure the Datadog Agent across your fleet of managed Macs, with no manual steps on each device. The script performs the same actions as the manual install, but runs in the background when the policy triggers.
```

1. Copy the provided installation command beginning with `sudo DD_API_KEY=`.
1. Open {{< ui >}}Terminal{{< /ui >}}. Find it in {{< ui >}}Applications{{< /ui >}} > {{< ui >}}Utilities{{< /ui >}} > {{< ui >}}Terminal{{< /ui >}}, or search for it with Spotlight (<kbd>⌘</kbd> + <kbd>Space</kbd>).
1. Paste the installation command into the terminal and run it. Enter your Mac password when prompted.

    The script installs the Agent, creates the `_dd-agent` system user, and registers the Agent as a background launch daemon. Installation takes one to two minutes.

### Verify the installation

To confirm that the Agent is running, in Datadog, go to [**Infrastructure** > **End User Devices**][104]. Your device appears within 5-10 minutes. If it does not appear after 10 minutes, verify your API key and confirm that the configuration was saved and the Agent was restarted.

Alternatively, you can run the following command in Terminal to verify the installation:

```shell
sudo datadog-agent status
```

In the output, confirm the following:
- The Agent version is 7.80 or later.
- **Status** is `Running`.
- `infrastructure_mode: end_user_device` is set.

> ### Enable Network Path (optional)

Reviewer 2: @evazorro can we take network path out of the docs for now? We're working through some enablement/billing issues and this is causing confusion.

Author: @eho1307 yup! Just pushed https://github.com/DataDog/documentation/pull/38311/commits/fadfc75dd6b1e3e5d8c9db8c8f2878b802933532 to remove all 4 network path sections (Mac/Windows manual/MDM)

        # tcp_method: <METHOD>

        ## @param traceroute_queries - integer - optional - default: 3
        ## Number of traceroutes to send for each check run.
        #
        # traceroute_queries: 3

        ## @param e2e_queries - integer - optional - default: 50
        ## Number of end-to-end probes to send for each check run.
        #
        # e2e_queries: 50

        ## @param tags - list of strings - optional
        ## A list of tags to attach to every metric and service check emitted by this instance.
        ##
        ## Learn more about tagging at https://docs.datadoghq.com/tagging
        #
        # tags:
        #   - <KEY_1>:<VALUE_1>
        #   - <KEY_2>:<VALUE_2>
    ```

1. Restart the Agent and system probe to start seeing network paths:

    ```shell
    sudo launchctl kickstart -k system/com.datadoghq.agent
    sudo launchctl kickstart -k system/com.datadoghq.sysprobe
    ```

[101]: /account_management/api-app-keys/
[102]: https://app.datadoghq.com/fleet/install-agent/latest?platform=macos
[103]: /agent/supported_platforms/osx/#commands
[104]: https://app.datadoghq.com/end-user-devices
[105]: /network_monitoring/network_path/setup/
{{% /tab %}}

> {{% tab "MDM deploy with Jamf Pro" %}}

Reviewer 2: ```suggestion
{{% tab "MDM deploy (Example with Jamf Pro)" %}}
```
I'd just like to call out that it's an example. Open to other ways of formatting this

Author: That works for me! Will tweak for sentence casing

---
title: Set up End User Device Monitoring on macOS
description: Deploy the Datadog Agent on macOS devices for End User Device Monitoring.
private: true
further_reading:
   - link: "/infrastructure/end_user_device_monitoring/"
     tag: "Documentation"
     text: "End User Device Monitoring"
   - link: "/infrastructure/end_user_device_monitoring/setup/"
     tag: "Documentation"
     text: "Set up End User Device Monitoring"
---

{{< callout url="https://www.datadoghq.com/product-preview/end-user-device-monitoring/" btn_hidden="false" >}}
End User Device Monitoring is in Preview. To enroll, click <b>Request Access</b>.
{{< /callout >}}

There are two ways to deploy the Datadog Agent on macOS devices for End User Device Monitoring. Choose the method that fits your environment:

- **Manual install**: Run a one-line script directly on the device. Best for testing a single machine or devices not managed by a mobile device management (MDM) solution.
- **MDM deploy**: Push the Agent to a fleet of managed Macs. This example uses a Jamf Pro policy and shell script to install the Agent in the background with no interaction on each device.

> Regardless of the method, the installer creates a dedicated system user named `ddagent`. This is expected: it allows the Agent to run as a background service with the appropriate system permissions. You do not need to manage this user.

Reviewer 3: ```suggestion
Regardless of the method, the installer creates a dedicated system user named `_dd-agent`. This is expected: it allows the Agent to run as a background service with the appropriate system permissions. You do not need to manage this user.
```

<div class="alert alert-danger">The <code>infrastructure_mode: end_user_device</code> setting is required. Without it, the device does not appear in the End User Devices view and is billed as a host.</div>

{{< tabs >}}
{{% tab "Manual install" %}}

Use this method to test a single machine or for devices that are not managed by an MDM solution.

### Prerequisites

- Administrator (`sudo`) access on the Mac.
- A valid Datadog API key. For instructions, see [API and Application Keys][101].
- macOS 11 or later.
- An active internet connection on the device.

### Install the Agent

1. In Datadog, navigate to [{{< ui >}}Fleet Automation{{< /ui >}} > {{< ui >}}Install Agents{{< /ui >}} > {{< ui >}}macOS{{< /ui >}}][102].
1. Click {{< ui >}}Select API Key{{< /ui >}} and choose an API key.
1. Copy the provided installation command beginning with `sudo DD_API_KEY=`.
1. Open {{< ui >}}Terminal{{< /ui >}}. Find it in {{< ui >}}Applications{{< /ui >}} > {{< ui >}}Utilities{{< /ui >}} > {{< ui >}}Terminal{{< /ui >}}, or search for it with Spotlight (<kbd>⌘</kbd> + <kbd>Space</kbd>).
1. Paste the installation command into the terminal and run it. Enter your Mac password when prompted.

>     The script installs the Agent, creates the `ddagent` system user, and registers the Agent as a background launch daemon. Installation takes one to two minutes.

Reviewer 3: ```suggestion
    The script installs the Agent, creates the `_dd-agent` system user, and registers the Agent as a background launch daemon. Installation takes one to two minutes.
```

Author: Thanks!

1. Launch the Datadog Agent Manager UI by running:

    ```shell
    sudo datadog-agent launch-gui
    ```

1. Navigate to {{< ui >}}Settings{{< /ui >}}.

> 1. Add a new parameter to the end of the YAML file:

Reviewer 3: Instead of launching the gui we should use the `DD_INFRASTRUCTURE_MODE="end_user_device"` arg during the running of the install script.

Author: This involves the same issue as this comment: https://github.com/DataDog/documentation/pull/38311#discussion_r3677072705

Namely, it would be better if the user could just copy the full command from Datadog and use it as-is. However, right now there's no way to toggle including that setting in the command.

We _can_ work around it in the docs, but it's the difference between

```
1. Choose your API key, then copy the command and run it.
```

and 

```
1. Choose your API key.
2. Copy this command, change the API key to the one you just picked (displayed
in the UI), and also change DDSITE if your Datadog site is not US-1.
```

cc @eho1307

Reviewer 2: We're a while away from adding it into Fleet so can we document it here?

Author: Updated. Let me know what you think of this version: 

* Mac: https://github.com/DataDog/documentation/pull/38311/changes#diff-a77c7337ba4a60e91d800d99482319873a12035c17b2422a7e6127f5b21d538fR41-R51 (also removed the GUI steps)
* Windows: https://github.com/DataDog/documentation/pull/38311/changes#diff-c49c703722f0ed578b9e292a38835de5d069d42ff03c4b70f655c48517b7c572R40-R55

Reviewer 2: Sorry @evazorro we decided to make the in-app change to simplify this guide. Can we revert it to point to the in-app instructions?

Author: Done! The preview is ready now:

* [Mac](https://docs-staging.datadoghq.com/eva/DOCS-14989-eudm-setup/infrastructure/end_user_device_monitoring/setup/macos/?tab=manualinstall#install-the-agent)
* [Windows](https://docs-staging.datadoghq.com/eva/DOCS-14989-eudm-setup/infrastructure/end_user_device_monitoring/setup/windows/?tab=manualinstall#install-the-agent)

- Devices running macOS 11 or later.
- Target devices enrolled in Jamf Pro with an active internet connection.

### Create the install script in Jamf

1. In the Jamf Pro console, go to **Settings** > **Computer Management** > **Scripts** and click **New**.
1. Give the script a name, such as `Datadog Agent EUDM`.
1. Paste the following into the **Script** field:

    ```shell
    #!/bin/zsh
    # Datadog Agent - End User Device Monitoring install script
    DD_API_KEY="$4"
    DD_SITE="datadoghq.com"

    DD_API_KEY="$DD_API_KEY" \
      DD_SITE="$DD_SITE" \
      DD_INFRASTRUCTURE_MODE="end_user_device" \
      bash -c "$(curl -L https://install.datadoghq.com/scripts/install_mac_os.sh)"

    echo "Datadog Agent installation complete."
    ```

    If your Datadog site is not US1, update `DD_SITE` to match. For the list of sites, see [Datadog sites][202].

1. To avoid storing your API key in the script body, pass it as a Jamf script parameter. In the **Options** tab, label parameter 4 as `DD_API_KEY`. The script reads this value from `$4`, and you enter the actual key when you [create the policy](#create-the-jamf-policy) in the next step.

>     The install script runs as root through Jamf, which provides the permissions needed to install the Agent and create the `ddagent` system user without user interaction.

Reviewer 3: ```suggestion
    The install script runs as root through Jamf, which provides the permissions needed to install the Agent and create the `_dd-agent` system user without user interaction.
```
